opensearch-operator/pkg/reconcilers/cluster.go

```diff
@@ -149,15 +149,69 @@ func (r *ClusterReconciler) reconcileNodeStatefulSet(nodePool opsterv1.NodePool,
 	// Fix selector.matchLabels (issue #311), need to recreate the STS for it as spec.selector is immutable
 	if _, exists := existing.Spec.Selector.MatchLabels["opensearch.role"]; exists {
 		r.logger.Info("deleting statefulset while orphaning pods to fix labels " + existing.Name)
-		opts := client.DeleteOptions{}
-		client.PropagationPolicy(metav1.DeletePropagationOrphan).ApplyToDelete(&opts)
-		if err := r.Delete(r.ctx, existing, &opts); err != nil {
-			r.logger.Info("failed to delete statefulset" + existing.Name)
+		if err := helpers.WaitForSTSDelete(r.ctx, r.Client, existing); err != nil {
+			r.logger.Error(err, "Failed to delete Statefulset for nodePool "+nodePool.Component)
```

Comment:
How do you detect un-healthy pod? is it a pod that is not on running status? or 0/1 is also un-healthy?

Reply:
The recovery process does not care for unhealthy pods, it just cares about the pods existing and assumes they will either start up and become healthy or kubernetes will take care of it.